hybrid/pkg/agent/agent.go

```diff
@@ -0,0 +1,169 @@
+package hybrid_agent
+
+import (
+	"bytes"
+	"context"
+	"reflect"
+	"strings"
+
+	hclog "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/hcl"
+	"github.com/hashicorp/hcl/hcl/printer"
+
+	"github.com/hewlettpackard/hybrid/pkg/common"
+	agentstorev1 "github.com/spiffe/spire-plugin-sdk/proto/spire/hostservice/server/agentstore/v1"
+	nodeattestorv1 "github.com/spiffe/spire-plugin-sdk/proto/spire/plugin/agent/nodeattestor/v1"
+	configv1 "github.com/spiffe/spire-plugin-sdk/proto/spire/service/common/config/v1"
+	"github.com/spiffe/spire/pkg/agent/plugin/nodeattestor/awsiid"
+	"github.com/spiffe/spire/pkg/agent/plugin/nodeattestor/azuremsi"
+	"github.com/spiffe/spire/pkg/agent/plugin/nodeattestor/gcpiit"
+	"github.com/spiffe/spire/pkg/agent/plugin/nodeattestor/k8spsat"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type HybridPluginAgent struct {
+	pluginList []common.Types
+	agentstorev1.UnimplementedAgentStoreServer
+	nodeattestorv1.UnsafeNodeAttestorServer
+	configv1.UnsafeConfigServer
+	logger      hclog.Logger
+	interceptor AgentInterceptorInterface
+	initStatus  error
+}
+
+func New() *HybridPluginAgent {
+	interceptor := new(HybridPluginAgentInterceptor)
+	return &HybridPluginAgent{interceptor: interceptor}
+}
+
+func (p *HybridPluginAgent) SetLogger(logger hclog.Logger) {
+	p.logger = logger
+
+}
+func (p *HybridPluginAgent) AidAttestation(stream nodeattestorv1.NodeAttestor_AidAttestationServer) error {
+	if len(p.pluginList) == 0 || p.initStatus != nil {
+		return status.Errorf(codes.InvalidArgument, "Plugin initialization error")
+	}
+
+	p.interceptor.setCustomStream(stream)
+	p.interceptor.SetContext(stream.Context())
+	p.interceptor.SetLogger(p.logger)
+
+	interceptors := []AgentInterceptorInterface{}
+
+	for i := 0; i < len(p.pluginList); i++ {
+		var newInterceptor AgentInterceptorInterface = p.interceptor.SpawnInterceptor()
+		newInterceptor.SetPluginName(p.pluginList[i].PluginName)
+		interceptors = append(interceptors, newInterceptor)
+
+		elem := reflect.ValueOf(p.pluginList[i].Plugin)
+		result := elem.MethodByName("AidAttestation").Call([]reflect.Value{reflect.ValueOf(newInterceptor)})
+		err := result[0].Interface()
+		if err != nil {
+			return status.Errorf(codes.Internal, "An error ocurred when during AidAttestation.")
```

Comment:
```suggestion
			return status.Errorf(codes.Internal, "An error ocurred during AidAttestation.")
```
also add plugin name (an error string, if possible) to the error message to help debugging